test/SILOptimizer/definite_init_diagnostics.swift

```diff
@@ -432,18 +432,18 @@ class DelegatingCtorClass {
   }
   
   convenience init(x: EmptyStruct, y: EmptyStruct) {
-    _ = ivar       // expected-error {{'self' used in property access 'ivar' before 'self.init' call}}
-    ivar = x       // expected-error {{'self' used in property access 'ivar' before 'self.init' call}}
+    _ = ivar       // expected-error {{'self' used before 'self.init' call}}
+    ivar = x       // expected-error* {{'self' used before 'self.init' call}}
```

Comment:
Definite initialization for convenience inits now uses the same code path as for delegating value type initializers, so some of the diagnostics specialized for partially-initialized class instances no longer fire. I can look into the QoI regressions after, but they're consistent with what you get for a value type delegating initializer.

Reply:
That seems wrong to me, then. A value type initializer is allowed to initialize a few fields and then throw those values away, but a convenience initializer isn't.

Either way, a test case should never use `*` on  an expected-error it's actually trying to test. Either the error is present or it's not.

Reply:
I agree; like I said, I'd like to look into the diagnostic issues once the codegen changes are in, because they also affect value type initializers today.

Reply:
I'm not happy with this, because this isn't just a diagnostic issue; it's a language change. Convenience initializers inherently *can't* do everything value initializers can. But I see the point about not having a long-lived branch that changes a massive number of tests.

Reply:
There's no language change, only the diagnostics changed.

Reply:
To be specific, "initialize a few fields and then throw it away" still does not work on convenience initializers because accessing any field of a class requires indirection through the instance pointer, and that pointer now does not exist until a self.init call happens.